tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -1,170 +1,154 @@
 import * as React from 'react';
+import userEvent from '@testing-library/user-event';
 
-import Button from 'amo/components/Button';
-import ConfirmButton, {
-  ConfirmButtonBase,
-  extractId,
-} from 'amo/components/ConfirmButton';
-import ConfirmationDialog from 'amo/components/ConfirmationDialog';
-import {
-  applyUIStateChanges,
-  createFakeEvent,
-  dispatchClientMetadata,
-  fakeI18n,
-  shallowUntilTarget,
-} from 'tests/unit/helpers';
+import ConfirmButton, { extractId } from 'amo/components/ConfirmButton';
+import { render as defaultRender, screen } from 'tests/unit/helpers';
 
 describe(__filename, () => {
-  const getProps = ({ i18n = fakeI18n(), ...props } = {}) => {
-    return {
-      i18n,
-      id: 'Collection-confirm-delete',
-      message: 'some warning message',
-      onConfirm: sinon.stub(),
-      store: dispatchClientMetadata().store,
-      ...props,
-    };
-  };
+  const defaultChildText = 'the default text of this button';
 
-  const render = ({ children, ...otherProps } = {}) => {
-    return shallowUntilTarget(
-      <ConfirmButton {...getProps(otherProps)}>
-        {children || 'the default text of this button'}
+  const render = ({
+    children = defaultChildText,
+    message = 'some warning message',
+    id = 'Some-confirm-button',
+    onConfirm = jest.fn(),
+    ...otherProps
+  } = {}) => {
+    return defaultRender(
+      <ConfirmButton
+        onConfirm={onConfirm}
+        message={message}
+        id={id}
+        {...otherProps}
+      >
+        {children}
       </ConfirmButton>,
-      ConfirmButtonBase,
     );
   };
 
-  const renderWithDialog = ({
-    store = dispatchClientMetadata().store,
-    ...otherProps
-  } = {}) => {
-    const root = render({ store, ...otherProps });
+  const renderWithDialog = ({ ...otherProps } = {}) => {
+    render(otherProps);
 
     // Click to open ConfirmationDialog.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find(ConfirmationDialog)).toHaveLength(1);
-
-    return root;
-  };
-
-  const getDialogProp = (root, propName) => {
-    const confirmDialog = root.find(ConfirmationDialog);
-    expect(confirmDialog).toHaveProp(propName);
-    return confirmDialog.prop(propName);
+    userEvent.click(
+      screen.getByRole('button', {
+        name: defaultChildText,
+      }),
+    );
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
```

Comment:
You'll notice I added a bunch of comments about not using `getByClassName` unless necessary. In this case it *could* be considered okay because we can configure the component to have a different message, and the buttons to have different text, so we can't guarantee to be able to assert about either of those. 

Having said that, there does appear to be one thing we could assert about that should always be true: The dialog will always have two buttons, so we could replace this assertion with `expect(screen.getAllByRole('button')).toHaveLength(2)`.